src/core/echarts.ts

```diff
@@ -1489,11 +1489,13 @@ class ECharts extends Eventful<ECEventDefinition> {
             }
 
             // If dispatchAction before setOption, do nothing.
-            ecModel && ecModel.eachComponent(condition, function (model) {
+            ecModel && ecModel.eachComponent(condition, function (model, modelIndex) {
                 if (!excludeSeriesIdMap || excludeSeriesIdMap.get(model.id) == null) {
                     if (isHighDownPayload(payload)) {
                         if (model instanceof SeriesModel) {
-                            if (payload.type === HIGHLIGHT_ACTION_TYPE && !payload.notBlur) {
+                            // only blur once
+                            const isFirstTimeBlur = modelIndex === 0;
```

Comment:
In some cases `modelndex` may not be `0`. It the index in the option. For example: if we change the test code to:
```ts
chart.dispatchAction({
  type: 'highlight',
  seriesIndex: [1],
 });
```

The `modelIndex` of the first iteration will be `1`. And no series will be blurred, which is not expected.

I think a better solution is using a flag to check if any series triggers blur.


There is another situation I think we need to consider:
If we set `blurScope` to `series`. Only other data in the same series will be blurred. Should we still use the "only one series will trigger blur" policy?

Reply:
Another unrelated suggestion: 

I think the old case you are using can be added to `hoverFocus.html`. It's more clear.

Reply:
@pissang Hi~
> In some cases modelndex may not be 0. It’s the index in the option. 

I have changed to flag implementation now~
(I was thinking that the second params `index` always starts with zero like `Array.forEach`.  )

> There is another situation I think we need to consider:
If we set blurScope to series. Only other data in the same series will be blurred. Should we still use the "only one series will trigger blur" policy?

I think when `blurScope === 'series'`, we can stick to the original _blur all_ logic.

> I think the old case you are using can be added to hoverFocus.html. It's more clear.

👌   I have moved the old testcase into the `hoverFocus.html`.

**When blurScope="series"**
![Jun-23-2021 17-26-08](https://user-images.githubusercontent.com/10973821/123072649-4c6edd80-d448-11eb-93bb-138b47c59103.gif)


**When blurScope!="series"**
![Jun-23-2021 17-24-51](https://user-images.githubusercontent.com/10973821/123072683-5264be80-d448-11eb-9b08-e06d304bfbd3.gif)

Reply:
> (I was thinking that the second params index always starts with zero like Array.forEach. )

Yeah, this API design may be misleading.

> I think when blurScope === 'series', we can stick to the original blur all logic.

There is another option `blurScope: 'coordinateSystem'` may have a similar issue. I'm thinking if we can support multiple series in https://github.com/apache/echarts/blob/4e671b6c95c1fe9606d2cdbdd4980a54ea47665c/src/util/states.ts#L514 and https://github.com/apache/echarts/blob/4e671b6c95c1fe9606d2cdbdd4980a54ea47665c/src/util/states.ts#L408 . Perhaps it's a better place to do this complex logic.

BTW: in the code of `blurSeries`, this comment can be removed to avoid confusion.
```ts
 // TODO blurScope: coordinate system
```

Reply:
@pissang I moved the logic into `echarts/src/util/states.ts`; 
(ps: the code modification in `src/core/echarts.ts` now is to improve code readability)

And for `blurScope: 'coordinateSystem'`, a new testcase is added in `hoverFocus.html`

![coordinate](https://user-images.githubusercontent.com/10973821/123208471-bb544100-d4f1-11eb-9d00-c6ba137378c7.gif)